frontend/src/pages/login.js

```diff
@@ -0,0 +1,98 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { useDispatch } from "react-redux";
+import { Link } from "react-router-dom";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+
+import InputField from "../components/input";
+import { login } from "../store/user";
+import { validate, validateField, getErrorPayload } from "../utils/validation";
+import * as userService from "../services/userService";
+
+const Login = () => {
+  const userLogin = useSelector((state) => state.login.value);
+  const dispatch = useDispatch();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const errors = validate(userLogin);
+    const payload = { ...userLogin, errors };
+    dispatch(login(payload));
+
+    // if there are no errors, try login through api
+    if (Object.keys(errors).length === 0) {
+      try {
+        await userService.login(userLogin);
+        const payload = { ...userLogin, success: true };
+        dispatch(login(payload));
```

Comment:
1. is this for users page only? is the admin side will be separated?
2. After login, it should redirect to home/dashboard. Don't show the login success text only.